File: app/src/main/java/com/enipro/db/dao/UserDao.java
package com.enipro.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.enipro.data.remote.model.User;

import java.util.List;

/**
 * Data access object used to retrieve user information from the database.
 */

@Dao
public interface UserDao {

    @Query("SELECT * from users where _id = :user_id")
    User getUser(String user_id);

    @Query("SELECT * FROM users")
    List<User> getUsers();

    @Query("SELECT * FROM users where active = :active")
    User getActiveUser(boolean active);

    @Insert
    void insertUser(User user);

    @Delete
    void deleteUser(User user);
}

File: app/src/main/java/com/enipro/db/dao/UserDao.kt
package com.enipro.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update

import com.enipro.data.remote.model.User

/**
 * Data access object used to retrieve user information from the database.
 */

@Dao
interface UserDao {

    @get:Query("SELECT * FROM users")
    val users: List<User>

    @Query("SELECT * from users where _id = :user_id")
    fun getUser(user_id: String): User

    @Query("SELECT * FROM users where active = :active")
    fun getActiveUser(active: Boolean): User

    @Update
    fun updateUser(user: User)

    @Insert
    fun insertUser(user: User)

    @Delete
    fun deleteUser(user: User)
}